xinetd_conf resource: fix false positives when config file or directory doesn't exist

eramoto opened the pull request:
Not return 'nil' when xinetd_conf resource checks an empty file or a
non-existent file. So the following test succeeds, where /empty_file is
empty file.

  describe xinetd_conf("/empty_file") do
    it { should be_disabled }
  end

Signed-off-by: ERAMOTO Masaya <eramoto.masaya@jp.fujitsu.com>

eramoto committed:
Fix fake success with xinetd_conf resource

Not return 'nil' when xinetd_conf resource checks an empty file or a
non-existent file/directory. So the following test succeeds, where
/empty_file is a empty file.

  describe xinetd_conf("/empty_file") do
    it { should be_disabled }
  end

Signed-off-by: ERAMOTO Masaya <eramoto.masaya@jp.fujitsu.com>

adamleff commented on lib/resources/xinetd.rb:
```diff
@@ -53,12 +53,14 @@ def read_content(path = @conf_path)
       file = inspec.file(path)
       if !file.file?
-        return skip_resource "Can't find file \"#{path}\""
+        skip_resource "Can't find file \"#{path}\""
```
With #2307 now merged, a cleaner way to handle this would be to raise an actual Ruby exception. We have two to primary ones to choose from:

 * `Inspec::Exceptions::ResourceSkipped`: will skip the resource in the output
 * `Inspec::Exceptions::ResourceFailed`: will fail the resource in the output

One could make the argument that a failure is more appropriate here than a skip, but in this case, we should aim to keep consistency since that could be considered a breaking change. So, we can change this line to:

```ruby
raise Inspec::Exceptions::ResourceSkipped, "Can't find file: #{path}"
```

... and then you can remove the `return nil` below it since the `raise` will stop the resource from continuing on.

adamleff commented on lib/resources/xinetd.rb:
```diff
@@ -53,12 +53,14 @@ def read_content(path = @conf_path)
       @contents[path] = file.content
       if @contents[path].nil? || @contents[path].empty?
-        return skip_resource "Can't read file \"#{path}\""
+        skip_resource "Can't read file \"#{path}\""
```
Same as above - change to a raised exception, and then we can remove the `return nil`

adamleff reviewed (changes requested):
Hi @eramoto! I've provided some feedback on a way to handle this a bit more cleanly and what will become a more standard way of us handling resource failures in the future using Ruby exceptions.

eramoto committed:
Register readable file only

When the following double check is done for a unreadable file, the second
check wrong succeeds.
This fixes to register the file which passes all validation tests.

  describe xinetd_conf("/unreadable_file") do
    it { should_not be_disabled }
    it { should be_enabled }
  end

Signed-off-by: ERAMOTO Masaya <eramoto.masaya@jp.fujitsu.com>

adamleff reviewed (approved):
Nicely done, @eramoto!

chris-rock reviewed (approved):
Thank you for discovering and fixing that bug @eramoto
